libs/remix-simulator/src/vm-context.ts

```diff
@@ -181,14 +201,28 @@ export class VMContext {
     } else
       stateManager = new StateManagerCommonStorageDump()
 
-    const common = new Common({ chain: 'mainnet', hardfork })
-    const blockchain = new (Blockchain as any)({ common })
+    const consensusType = hardfork === 'berlin' || hardfork === 'london' ? ConsensusType.ProofOfWork : ConsensusType.ProofOfStake
```

Comment:
Is this check sufficient? How about other hardforks like istanbul?

Reply:
currently we instanciate only "berlin" and "london", so this is the last two on proof of stake fork we are still using.